src/tree/Element.mjs

```diff
@@ -1411,6 +1412,24 @@ export default class Element {
         }
     }
 
+    get collision() {
+        return this._collision;
+    }
+
+    set collision(v) {
+        this._collision = v;
+    }
+
+    _updateCollision() {
+        if (this.collision && this.__parent && this.__parent.collision === undefined) {
+            /**
+             * Mark collision as 2 to indicate atleast one descendant has collision enabled.
+             * This narrows down the search for elements with active collision.
+            */
+            this.__parent.collision = 2;
```

Comment:
Any reasoning behind marking it as `2` Other then: https://github.com/rdkcentral/Lightning/blob/6dfd288945b6207f55391262db86e66e85381d77/src/application/Application.mjs#L564 evaluating to true?

Reply:
Currently, it allows to filter out all the unnecessary elements early on before further evaluation in response to mouse events (if the elements were evaluated to `true` instead they would be needlessly processed). Other than that, this distinction might become useful later should this feature need any alterations.